process_manager/views/partials.py

```diff
@@ -96,10 +96,17 @@ def process_table(request: HttpRequest) -> HttpResponse:
 @login_required
 def messages(request: HttpRequest) -> HttpResponse:
     """Renders Kafka messages from the database."""
+    search = request.GET.get("search", "").lower()
+
     messages = []
     for msg in DruncMessage.objects.all():
+        # Filter messages based on search parameter.
+        if search not in msg.message.lower():
+            continue
+
```

Comment:
Can make this simpler and more efficient with `DruncMessage.objects.filter(message__icontains=search)`.